diff --git a/lib/src/takion.c b/lib/src/takion.c
index f57f9ffc..e344fd10 100644
--- a/lib/src/takion.c
+++ b/lib/src/takion.c
@@ -15,8 +15,11 @@
 #include <assert.h>
 
 #ifdef __APPLE__
+#include <TargetConditionals.h>
+#if TARGET_OS_OSX
 #include <CoreServices/CoreServices.h>
 #endif
+#endif
 
 #ifdef _WIN32
 #include <ws2tcpip.h>
@@ -253,7 +256,7 @@ CHIAKI_EXPORT ChiakiErrorCode chiaki_takion_connect(ChiakiTakion *takion, Chiaki
 			goto error_sock;
 		}
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && TARGET_OS_OSX
 		SInt32 majorVersion;
 		Gestalt(gestaltSystemVersionMajor, &majorVersion);
 		if(majorVersion < 11)
@@ -342,7 +345,7 @@ CHIAKI_EXPORT ChiakiErrorCode chiaki_takion_connect(ChiakiTakion *takion, Chiaki
 		}
 		if(info->ip_dontfrag)
 		{
-#if defined(__APPLE__)
+#if defined(__APPLE__) && TARGET_OS_OSX
 			SInt32 majorVersion;
 			Gestalt(gestaltSystemVersionMajor, &majorVersion);
 			if(majorVersion < 11)
